manifests/service.pp

```diff
@@ -5,42 +5,48 @@
 
   systemd::unit_file { 'pulpcore-api.socket':
     content => template('pulpcore/pulpcore-api.socket.erb'),
-    active  => true,
-    enable  => true,
+    active  => $pulpcore::service_ensure,
+    enable  => $pulpcore::service_enable,
   }
 
   systemd::unit_file { 'pulpcore-api.service':
     content => template('pulpcore/pulpcore-api.service.erb'),
-    active  => true,
-    enable  => true,
+    active  => $pulpcore::service_ensure,
+    enable  => $pulpcore::service_enable,
   }
 
   systemd::unit_file { 'pulpcore-content.socket':
     content => template('pulpcore/pulpcore-content.socket.erb'),
-    active  => true,
-    enable  => true,
+    active  => $pulpcore::service_ensure,
+    enable  => $pulpcore::service_enable,
   }
 
   systemd::unit_file { 'pulpcore-content.service':
     content => template('pulpcore/pulpcore-content.service.erb'),
-    active  => true,
-    enable  => true,
+    active  => $pulpcore::service_ensure,
+    enable  => $pulpcore::service_enable,
   }
 
   systemd::unit_file { 'pulpcore-resource-manager.service':
     content => template('pulpcore/pulpcore-resource-manager.service.erb'),
-    active  => true,
-    enable  => true,
+    active  => $pulpcore::service_ensure,
+    enable  => $pulpcore::service_enable,
   }
 
   systemd::unit_file { 'pulpcore-worker@.service':
     content => template('pulpcore/pulpcore-worker@.service.erb'),
   }
 
+  if $pulpcore::service_ensure == false {
+    $worker_ensure = 'stopped'
+  } else {
+    $worker_ensure = 'running'
+  }
```

Comment:
I'm never a fan of checking for negative first so I'd normally write:
```suggestion
  if $pulpcore::service_ensure {
    $worker_ensure = 'running'
  } else {
    $worker_ensure = 'stopped'
  }
```

You can also simplify it:
```suggestion
  $worker_ensure = bool2str($pulpcore::service_ensure, 'running', 'stopped')
```

Reply:
Oh, even easier, you can actually pass a boolean to `ensure` so no need to convert that. If you do that, then I'm fine with accepting a boolean in the class.